pkg/skaffold/build/local/local.go

```diff
@@ -42,8 +43,25 @@ func (b *Builder) Build(ctx context.Context, out io.Writer, tags tag.ImageTags,
 	}
 	defer b.localDocker.Close()
 
+	pruner := newPruner(b.localDocker, b.pruneChildren)
+
+	if b.prune {
+		pruner.startCleanupOldImages(ctx, out, artifacts)
+	}
+
 	builder := build.WithLogFile(b.buildArtifact, b.muted)
-	return build.InParallel(ctx, out, tags, artifacts, builder, *b.cfg.Concurrency)
+
+	rt, err := build.InParallel(ctx, out, tags, artifacts, builder, *b.cfg.Concurrency)
+
+	if b.prune {
+		if b.mode == config.RunModes.Dev {
+			pruner.startCleanupOldImages(ctx, out, artifacts)
```

Comment:
Can you explain why in devMode we clean up images asynchronously ?
Is it because docker can run out of memory multiple times in dev mode?

Can you please add the implementation in the PR description e.g
```
This Pr previous versions of docker images from the local registry, when local docker builder is used if `cleanup=true` flag is set. 

Pruning images can happen asynchronously and synchronously.
In dev mode, skaffold will ....

For other modes like ....
```

Reply:
Oruning can be time consuming, so we start it in a separate go-routine. It's safe as we don't need old images for build. 
On the other hand, in 'build' mode this pruning is the last action and we need to complete it before we exit, that's why we run is synchronously.

I'll update the PR description